libr/asm/asm.c

```diff
@@ -307,8 +307,8 @@ static int has_bits(RAsmPlugin *h, int bits) {
 R_API void r_asm_set_cpu(RAsm *a, const char *cpu) {
 	if (a) {
 		free (a->cpu);
+		a->cpu = cpu? strdup (cpu): NULL;
```

Comment:
r_str_dup checks for null, so you can skip a ternary op here

Reply:
I'll make a patch.